### ISSUE TITLE
Query

### ISSUE THREAD

**Issue by ankitpatnala** [vturrisi/solo-learn#128]
May be very stupid question but i found you always used class loss along with main loss function which is mentioned in paper. Class loss is CE loss between targets and predicted logits.  
is it something supervised contrastive learning or  am I reading the code wrong.

**Comment by ankitpatnala** [comment]
Just a weird question. 
How does it predict logits when target is not defined for the problem(self-supervised) ?
what i mean may be in real target dog is class 1 but in our self-supervised model, it could be class 2.

https://github.com/vturrisi/solo-learn/blob/4c195ca066a24ad6569f713f65137c8e1117a839/solo/methods/base.py#L401

i know it works but don't know how it works ?

**Comment by vturrisi** [comment]
hey,
the dataset classes expect that your data is organized in sub-folders (one for each class), so in the targets you actually have the right label

**Comment by ankitpatnala** [comment]
One more thing. I am trying to install umap via  pip resources (pip install umap) where the library doesn't have the functionalitites used in the codebase.  There is a library named umap-learn which matches to the code used in auto_umap.py. Can you please confirm about it ?

**Comment by DonkeyShot21** [comment]
Hi, you can find all the required packages for using UMAP at this line:
https://github.com/vturrisi/solo-learn/blob/4c195ca066a24ad6569f713f65137c8e1117a839/setup.py#L8

Also, the best way to install `solo-learn` with UMAP support is to just run:
```
pip3 install .[umap]
```
